src/silksong_memory.rs

```diff
@@ -674,6 +680,60 @@ impl Default for SceneStore {
 
 // --------------------------------------------------------
 
+pub fn get_tools_version(mem: &Memory, pd: &PlayerDataPointers) -> Option<i32> {
+    mem.deref(&pd.tools_version).ok()
+}
+
+pub fn find_tool(tool_utf16: &[u16], mem: &Memory, pd: &PlayerDataPointers) -> bool {
```

Comment:
If `find_tool` returned an `Option`, even just an `Option<()>`, then some of the code below could take advantage of the `?` question mark